### PR TITLE
support cocktail images in firebase & json + add recipe fragment

### PR DESCRIPTION
**Pull request by zoharhen**
not finished at all, will fix it today
fragment is still corrupted but included image+ingredients+steps blocks.
still missing: TTS (text-to-speech), link to share recipe+app, buttons like favorite, delete, edit etc.

### CODE REVIEW

File: app/src/main/java/com/example/cocktails/ScrollingActivity.kt [hunk 1]
```diff
@@ -35,8 +35,8 @@ import kotlinx.android.synthetic.main.filter_dialog.view.*
 
 @Parcelize
 data class Cocktail(val name: String, val type: String, val steps: Array<String>,
-                    val ingredients: Array<String>, val image: String, val isCustom: Boolean = false,
-                    val glass: Uri?): Parcelable
+                    val ingredients: Array<String>, val clipart: String, val image: String,
+                    val isCustom: Boolean = false, val glass: String?): Parcelable
```
Comment (zoharhen) [inline app/src/main/java/com/example/cocktails/ScrollingActivity.kt:39]: @shahafhermann temporary change, feel free to re rewrite it, but notice i added new clipart and image parameters as well

File: app/src/main/java/com/example/cocktails/ScrollingActivity.kt [hunk 2]
```diff
@@ -55,15 +55,15 @@ class ScrollingActivity : AppCompatActivity() {
...
-        mAuth.signInAnonymously()
-            .addOnCompleteListener { task ->
-                if (task.isSuccessful) {
-                    val user: FirebaseUser? = mAuth.currentUser
-                    // todo: einav - get user custom cocktails objects? (remember to save the object once being created). maybe use SP instead?
-                } else {
-                    Toast.makeText(applicationContext, "Error loading data", Toast.LENGTH_SHORT).show()
-                }
-            }
+//        mAuth.signInAnonymously()
```
Comment (zoharhen) [inline app/src/main/java/com/example/cocktails/ScrollingActivity.kt:58]: @enavhasson your call if you want to use mAuth (firebase authentication), if not - you should delete all mAuth  references (here and via gardel and 'Cocktails' classes) :)

File: app/src/main/java/com/example/cocktails/ui/main/RecipeFragment.kt [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    @Override
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        rootView = inflater.inflate(R.layout.recipe_item, container, false)
+        this.initViews()
+        return rootView
+    }
+
+    private fun initViews() {
+        (activity?.applicationContext as Cocktails).mStorageRef.child("cliparts/" + cocktail.clipart + ".png")
+            .downloadUrl.addOnSuccessListener { img ->
```
Comment (zoharhen) [inline app/src/main/java/com/example/cocktails/ui/main/RecipeFragment.kt:51]: @zoharhen my bug: fix it to "images" directory instead

File: app/src/main/java/com/example/cocktails/ui/main/RecipeFragment.kt [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                        .into(rootView.findViewById(R.id.iv_cocktail))
+                }
+            }
+        val ingredients = rootView.findViewById<TextView>(R.id.ingredientContent)
+        ingredients.text = cocktail.ingredients.joinToString(separator = "\n")
+        this.initPreparationSection()
+
+        val preparation = rootView.findViewById<TextView>(R.id.preparationSetion)
+        preparation.setOnHoverListener { v, _ ->
+            TooltipCompat.setTooltipText(v,"Click a step in order to play the text.\n Press a long click in order to mark a step as done"
```
Comment (zoharhen) [inline app/src/main/java/com/example/cocktails/ui/main/RecipeFragment.kt:64]: @zoharhen my bug: fix tooltip

File: app/src/main/java/com/example/cocktails/ui/main/RecipeFragment.kt [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        val recyclerViewPreparation =  rootView.findViewById(R.id.recyclerPreparation) as RecyclerView
+        mAdapterPreparation = context?.let { PreparationAdapter(it, generatePreparation()!!, this) }
+        val mLayoutManagerPreparation = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
+        recyclerViewPreparation.layoutManager = mLayoutManagerPreparation
+        recyclerViewPreparation.itemAnimator = DefaultItemAnimator()
+        recyclerViewPreparation.adapter = mAdapterPreparation
+    }
+
+    override fun onItemClicked(position: Int) {
+        TODO("Not yet implemented")
```
Comment (zoharhen) [inline app/src/main/java/com/example/cocktails/ui/main/RecipeFragment.kt:85]: @zoharhen  mine: add TTS support

File: app/src/main/java/com/example/cocktails/ui/main/SectionsPagerAdapter.kt [hunk 6]
```diff
@@ -16,18 +16,16 @@ private val TAB_TITLES = arrayOf(
...
 
     override fun getItem(position: Int): Fragment {
         // getItem is called to instantiate the fragment for the given page.
         if (position == 0) { // Recipe tab
-            // return a recipe fragment
+            return RecipeFragment.newInstance(cocktail) as Fragment
         }
         else if (position == 1) {  // AR tab
-            return ARFragment(cocktail)
+//            return ARFragment(cocktail)
```
Comment (zoharhen) [inline app/src/main/java/com/example/cocktails/ui/main/SectionsPagerAdapter.kt:28]: @shahafhermann  I changed it by accident, notice when you push your code :)

File: app/src/main/java/com/example/cocktails/ScrollingActivity.kt (hunk 2 above)
Comment (enavhasson) [reply app/src/main/java/com/example/cocktails/ScrollingActivity.kt:58]: perfect

File: app/src/main/java/com/example/cocktails/ScrollingActivity.kt (hunk 1 above)
Comment (shahafhermann) [reply app/src/main/java/com/example/cocktails/ScrollingActivity.kt:39]: No problem, that's probably the best solution.

File: app/src/main/java/com/example/cocktails/ui/main/SectionsPagerAdapter.kt (hunk 6 above)
Comment (shahafhermann) [reply app/src/main/java/com/example/cocktails/ui/main/SectionsPagerAdapter.kt:28]: Thanks for the heads up

**Pull request merged**